en/computer_vision/visual_inertial_odometry.md

```diff
@@ -0,0 +1,138 @@
+It is commonly used to navigate a vehicle in situations where GPS is absent or unreliable (e.g. indoors, or when flying under a bridge).
+
+VIO uses [Visual Odometry](https://en.wikipedia.org/wiki/Visual_odometry) to estimate vehicle *pose* from camera images, combined with inertial measurements from the vehicle IMU (to correct for errors associated with rapid vehicle movement resulting in poor image capture).
+
+This topic shows how to set up PX4 and a companion computer to use the *supported* VIO setup.
+
+{% youtube %}
+https://youtu.be/gWtrka2mK7U
+{% endyoutube %}
+
+> **Tip** The [Auterion product video](https://auterion.com/enabling_uav_navigation_in_environments_with_limited_or_no_gps_signal/) above shows a vehicle flying using the [supported setup](#supported_setup).
+
+<span></span>
+> **Note** This (supported) solution uses ROS for routing VIO information to PX4.
+  PX4 itself does not care about the source of messages, provided they are provided via the appropriate [MAVLink Interface](https://dev.px4.io/master/en/ros/external_position_estimation.html#px4-mavlink-integration).
+
+
+## Supported Setup {#supported_setup}
+
+The supported setup uses the [T265 Intel Realsense Tracking Camera](../peripherals/camera_t265_vio.md) and ROS (running on a companion computer) to supply odometry information to PX4.
+The Auterion [VIO bridge ROS node](https://github.com/Auterion/VIO_bridge) provides a bridge between the camera and ROS.
+
+
+
+### Camera Mounting
+
+Attach the camera to the companion computer and mount it to the frame:
+
+- Connect the [T265 Intel Realsense Tracking Camera](../peripherals/camera_t265_vio.md) using the supplied USB-C to USB-A connector.
+- Mount the camera with lenses pointing down if at possible (default).
+- The camera is very senstive to vibration; a soft mounting for vibration isolation is recommended.
+
+
+### ROS/VIO Setup
+
+These instructions explain how to set up VIO using ROS:
+- On the companion computer, install and configure [MAVROS](https://dev.px4.io/master/en/ros/mavros_installation.html).
+  Verify it is connected successfully to the flight controller.
+  > **Tip** You can use the *QGroundControl* [MAVLink Inspector](https://docs.qgroundcontrol.com/en/analyze_view/mavlink_inspector.html) to verify that you're getting `HEARTBEAT` messages from the companion computer.
+- Get the Auterion [VIO bridge ROS node](https://github.com/Auterion/VIO_bridge):
+  - Clone this repository in your catkin workspace.
+    ```
+	cd ~/catkin_ws/src
+	git clone https://github.com/Auterion/VIO.git
+	```
+  - Build the package:
+    ```
+	cd ~/catkin_ws/src
+	catkin build px4_realsense_bridge
+	```
+- Configure the *VIO bridge launch file* (typically [bridge_mavros.launch](https://github.com/Auterion/VIO/blob/master/launch/bridge_mavros.launch) or [bridge.launch](https://github.com/Auterion/VIO/blob/master/launch/bridge.launch) ) to reflect your camera orientation.
```

Comment:
@baumanta I looked at the launch files, and it still isn't clear how you would specify a different orientation than the default (lenses down). Can you give me an example - e.g. what would I paste in to the file if the camera was pointed 45 degrees facing forward?

Reply:
it is that line:
<node pkg="tf" type="static_transform_publisher" name="tf_baseLink_cameraPose"
        args="0 0 0 0 1.5708 0 base_link camera_pose_frame 1000"/>

This is a static transform that links the camera ROS frame `camera_pose_frame` to the mavros drone frame `base_link`. 

the first three numbers specify translation x,y,z (center of fcu to camera):
so if the camera is 10 cm in front of the pixhawk and 4 up, the first three numbers would be : [0.1, 0, 0.04,...]
And the next three numbers specify rotation (yaw, pitch, roll). 
So [... 0, 1.5708, 0] means pitch down by 90deg (facing the ground). facing straight forward would be [... 0 0 0].